mh/session.py

```diff
@@ -167,7 +167,9 @@ def get_layer_users(self, server_id, gate_id, city_id, first_index, count):
         return players[start:start+count]
 
     def leave_server(self):
-        DB.leave_server(self)
+        old_server = self.local_info["server_id"]
+        if old_server is not None:
```

Comment:
I think that this is `if` is unnecessary because since you are using the `[]` operator, if the dictionary doesn't find the key it would raise a `KeyError` instead of returning `None`. I think that if you want to get a default value (`None`) you need to use the `get` function

Reply:
I'm wrong, when session is constructed a value (`None`) is assigned to that key